lib/cn/kblock_reader.c

```diff
@@ -29,8 +29,10 @@
 static HSE_ALWAYS_INLINE bool
 kblock_hdr_valid(const struct kblock_hdr_omf *omf)
 {
-    return (HSE_LIKELY(
-        omf_kbh_magic(omf) == KBLOCK_HDR_MAGIC && omf_kbh_version(omf) <= KBLOCK_HDR_VERSION));
+    uint32_t vers = omf_kbh_version(omf);
+
+    return (HSE_LIKELY(omf_kbh_magic(omf) == KBLOCK_HDR_MAGIC &&
+                       vers >= KBLOCK_HDR_VERSION5 && vers <= KBLOCK_HDR_VERSION));
```

Comment:
Am I the only one that prefers range checks written in a form that matches the way you would write them on paper?  On paper I would write:   `5 <= vers <= 6`, which I would translate to `5 <= vers && vers <= 6`.  But most C coders write:  `vers >= 5 && vers <= 6`.  Harder to grok the meaning IMO.

Reply:
I agree with this statement

Reply:
Agreed, the on paper version reads clear.